float64 c0
float64 c1
float64 c2
